lib/util.cpp

```diff
@@ -625,3 +625,24 @@ double rand_normal() {
     cached = true;
     return z*cos(PI2*u2);
 }
+
+// determines the real path and filename of the current process
+// not the current working directory
+//
+int get_real_executable_path(char* path, size_t max_len) {
+#ifdef HAVE__PROC_SELF_EXE
+    int ret = readlink("/proc/self/exe", path, max_len);
+    if ( ret >= 0) {
+        path[ret] = '\0'; // readlink does not null terminate
```

Comment:
In the somewhat unlikely event that the link is `max_len` bytes long or longer this write goes one past the end of the buffer. Better have `max_len-1` with `readlink`.